guide/src/guide/new-cell.md

```diff
@@ -1 +1,221 @@
-# Write a New Cell (TODO)
+# Write a New Cell
+
+Following this guide, you can write a custom cell yourself.
+
+## Basic Structure
+
+We have some necessary structs and traits which you must impl for your own cell. We take a cell which loses one packet in every `n` packets (`n` can be set by users freely) as example to explain basic structure of a custom cell.
```

Comment:
`impl` to `implement`. Abbreviations should be avoided in the guide (you can use abbreviations in documents or comments).

You can use `drop` instead of `lose` for verb and `Lost` or `Drop` for the name of the cell.